### DISCUSSION TITLE
Como funcionam os blobs no JavaScript?

### DISCUSSION THREAD

**Discussion by felquis** [frontendbr/forum#380] [Perguntas e Respostas]
Isso é algo que sempre me assombra, e até hoje tive bastante dificuldade de entender qual é a moral dele.

Alguém poderia me dar uma luz, me lincar para alguma explicação bem para iniciantes?

https://developer.mozilla.org/en-US/docs/Web/API/Blob

**Comment by fdaciuk** [comment]
A ideia é que você possa usar blobs para representar arquivos grandes, em formato binária (_Binary Large OBjects_). No caso, você pode transformar qualquer tipo de arquivo em `Blob`, e usá-lo em memória, por ele ocupar menos espaço que um arquivo físico.

No JS, normalmente você cria um `Blob` a partir de um arquivo físico para fazer upload.

Algumas definições que podem tentar deixar mais claro você encontrar [aqui](http://searchsqlserver.techtarget.com/definition/BLOB) e [aqui](https://en.wikipedia.org/wiki/Binary_large_object) :)
